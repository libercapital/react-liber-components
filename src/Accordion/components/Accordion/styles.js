import styled from 'styled-components';
import { media } from '../../../Util';
import { handleThemeFromObject } from '../../../Themes';

export const AccordionBox = styled.div`
  overflow: hidden;
  width: 100%;
  background-color: ${props =>
    props.open
      ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
      : handleThemeFromObject(props.theme, 'accordion.boxBackgroundClosed', '#f9f9f9')};
  border: ${props =>
    props.open
      ? `1px solid ${handleThemeFromObject(props.theme, 'accordion.boxBorderOpen', '#fff')}`
      : `solid 1px ${handleThemeFromObject(
          props.theme,
          'accordion.boxBorderClosed',
          'rgba(150, 150, 150, 0.18)',
        )}`};
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  &:hover {
    background-color: ${props =>
      props.open
        ? handleThemeFromObject(props.theme, 'accordion.boxBackgroundOpen', '#fff')
        : handleThemeFromObject(props.theme, 'accordion.boxBackgroundHover', '#f3f3f3')};
  }
`;

export const ShowBox = styled.div`
  max-height: ${props => (props.open ? props.size : '0px')};
  transition: max-height 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  overflow: hidden;
`;

export const AccordionAction = styled.div`
  padding: 20px 50px;
  cursor: pointer;
  box-sizing: border-box;
  display: inline-block;
  background-color: ${props => (props.open ? props.color || 'transparent' : 'transparent')};
  width: 100%;
  transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  position: relative;

  ${media.large`
    padding: 30px 50px;
  `}

  &:after {
    position: absolute;
    top: calc(50% - 4px);
    left: 15px;
    width: 0;
    height: 0;
    padding: 0;
    content: '';
    border-left: 7px solid transparent;
    border-right: 7px solid transparent;
    border-top: 8px solid
      ${props =>
        props.open
          ? handleThemeFromObject(props.theme, 'accordion.actionBorderTopOpen', '#009dff')
          : handleThemeFromObject(props.theme, 'accordion.actionBorderTopClosed', '#4e4e4e')};
    pointer-events: none;
    transform: ${props => (props.open ? 'rotate(0deg)' : 'rotate(-90deg)')};
    transition: all 400ms ${props => (props.open ? 'ease-in' : 'ease-out')};
  }
`;
