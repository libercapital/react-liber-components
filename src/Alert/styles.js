import styled, { css } from 'styled-components';
import { MdClose } from '../Icons/MaterialDesign';

const icon = css`
  --size: 24px;
  min-width: var(--size);
  max-width: var(--size);
  min-height: var(--size);
  max-height: var(--size);
`;

export const CloseIcon = styled(MdClose)`
  ${icon}
  &:hover {
    cursor: pointer;
  }
`;

export const AlertBox = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
`;

export const AlertContent = styled.div`
  display: flex;
  flex-grow: 1;
  flex-direction: column;
  padding: 24px 0;
`;

export const AlertInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: 500;
  line-height: 1.38;
  letter-spacing: normal;
  color: #405f71;
`;
export const AlertAdditionalInfoContent = styled.div`
  font-family: Roboto;
  font-size: 16px;
  font-weight: normal;
  line-height: 1.38;
  letter-spacing: 0px;
  color: #587383;
  padding-top: 4px;
`;

export const AlertIcon = styled.div`
  display: flex;
  flex-grow: 0;
  align-items: center;
  padding: 24px;
`;
