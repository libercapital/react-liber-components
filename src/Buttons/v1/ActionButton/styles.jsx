import styled from 'styled-components';
import { media } from '../../../Util';


export const Label = styled.div`
  text-align: center;
  width: unset;
  opacity: 1;
  transition: all 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  white-space: nowrap;
  color: inherit;

  ${media.large`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
  ${media.extraLarge`
    width: ${props => (props.hover ? props.size || '0px' : '0px')};
    opacity: ${props => (props.hover ? '1' : '0')};
  `}
`;

export const Icon = styled.div`
  margin-right: 15px;
  transition: margin 300ms  ${props => (props.hover ? 'ease-out' : 'ease-in')};
  ${media.large`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
  ${media.extraLarge`
    margin-right: ${props => (props.hover ? '15px' : '0px')};
  `}
`;
