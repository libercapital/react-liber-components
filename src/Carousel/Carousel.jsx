import React, { Component } from 'react';
import PropTypes from 'prop-types';
import _ from 'lodash';
import Swipe from 'react-easy-swipe';
import {
  Hidden, Carousels, DotsContainer, Dot as DefaultDot, Body,
  LeftChavron as DefaultLeft, RightChavron as DefaultRight,
} from './Carousel.styles';


class Carousel extends Component {
  constructor(props) {
    super(props);
    this.state = {
      index: 0,
      size: React.Children.toArray(props.children).length, // eslint-disable-line
    };
  }

  static getDerivedStateFromProps(props, state) {
    const size = React.Children.toArray(props.children).length;
    if (state.size !== size) {
      return {
        size,
      };
    }
    return {};
  }

  componentDidMount() {
    this.startInterval();
  }

  componentWillUnmount() {
    if (this.interval) clearInterval(this.interval);
  }

  startInterval = () => {
    const { loop, loopTime, loopLeft } = this.props;
    if (loop) {
      this.interval = setInterval(loopLeft ? this.handlePrev : this.handleNext, loopTime);
    }
  }

  removeInterval = () => {
    if (this.interval) clearInterval(this.interval);
  }

  handleDotClick = ({ target }) => {
    const index = target.getAttribute('data-index');
    this.removeInterval();
    this.setState({ index: parseInt(index, 10) }, this.startInterval);
  }

  handleActionNext = () => {
    this.removeInterval();
    this.handleNext({}, this.startInterval);
  }

  handleActionPrev = () => {
    this.removeInterval();
    this.handlePrev({}, this.startInterval);
  }

  handleNext = (event, callback) => {
    this.setState(state => ({
      index: state.index + 1 >= state.size ? 0 : state.index + 1,
    }), callback);
  }

  handlePrev = (event, callback) => {
    this.setState(state => ({
      index: state.index - 1 <= -1 ? state.size - 1 : state.index - 1,
    }), callback);
  }


  render() {
    const {
      width, height, children, innerDots,
    } = this.props;
    const { index } = this.state;
    const Dot = this.props.dot;
    const Left = this.props.leftArrow;
    const Right = this.props.rightArrow;

    const dots = React.Children.toArray(children).map((child, i) => (
      <Dot data-index={i} key={_.uniqueId('carousel_child_')} current={index === i} onClick={this.handleDotClick} />
    ));
    const dotsContainer = (<DotsContainer inner={innerDots}>{dots}</DotsContainer>);

    return (
      <Body>
        <Swipe
          onSwipeRight={this.handleActionPrev}
          onSwipeLeft={this.handleActionNext}
        >
          <Hidden
            width={width}
            height={height}
            onMouseEnter={this.removeInterval}
            onMouseLeave={this.startInterval}
          >
            <Carousels size={width} index={index}>
              {children}
            </Carousels>
            <Left onClick={this.handlePrev} />
            {innerDots ? dotsContainer : null}
            <Right onClick={this.handleNext} />
          </Hidden>
        </Swipe>
        {!innerDots ? dotsContainer : null}
      </Body>
    );
  }
}

Carousel.propTypes = {
  children: PropTypes.arrayOf(PropTypes.element).isRequired,
  width: PropTypes.number,
  height: PropTypes.number,
  dot: PropTypes.oneOfType([PropTypes.any]),
  leftArrow: PropTypes.oneOfType([PropTypes.any]),
  rightArrow: PropTypes.oneOfType([PropTypes.any]),
  innerDots: PropTypes.bool,
  loop: PropTypes.bool,
  loopTime: PropTypes.number,
  loopLeft: PropTypes.bool,
};
Carousel.defaultProps = {
  width: 1200,
  height: 300,
  dot: DefaultDot,
  leftArrow: DefaultLeft,
  rightArrow: DefaultRight,
  innerDots: true,
  loop: false,
  loopTime: 4000,
  loopLeft: false,
};

export default Carousel;
