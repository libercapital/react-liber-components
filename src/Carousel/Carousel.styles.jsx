import styled, { css } from 'styled-components';
import { media } from '../Util';
import { handleThemeFromObject as getTheme } from '../Themes';
import { FaChevronDown } from '../Icons';


const getPos = (size, index) => {
  const pos = size && index ? -1 * size * parseFloat(index) : 0;
  return `${pos}px`;
};


export const Carousels = styled.div`
  display: flex;
  transform: translateX(${props => getPos(props.size, props.index)});
  transition: transform .5s ease-in-out;
  width: 100%;
  height: 100%;

  & > *{
    min-width: 100%;
    min-height: 100%;
  }
`;

const inner = css`
  position: absolute;
  bottom: 0px;
  left:0px;
  opacity: 0.2;
  transition: opacity .25s ease-in-out;
`;

const outer = css``;


export const DotsContainer = styled.div`
  min-height: 30px;
  width: 100%;

  display: flex;
  justify-content: center;
  align-items: center;
  ${props => (props.inner ? inner : outer)};
`;


export const Dot = styled.span`
  min-width: 10px;
  min-height: 10px;
  border-radius: 50%;
  border: 2px solid var(--color);
  margin: 0px 4px;
  background-color: ${props => (props.current ? 'var(--color)' : 'transparent')};
  cursor: pointer;
`;

const chavron = css`
  width: 50px;
  height: 50px;
  position: absolute;
  top: calc(50% - 25px);
  fill: var(--color);
  cursor: pointer;
  opacity: 0.2;
  transition: opacity .25s ease-in-out;
`;


export const LeftChavron = styled(FaChevronDown)`
  transform: rotate(90deg);
  left: 10px;
  ${chavron}

`;


export const RightChavron = styled(FaChevronDown)`
  transform: rotate(-90deg);
  right: 10px;
  ${chavron}
`;

export const Hidden = styled.div`
  --color: ${props => getTheme(props.theme, 'colors.v2.gray1', '#eaf7ff')};
  width: ${props => `${props.width}px` || '600px'};
  height: ${props => `${props.height}px` || '150px'};
  overflow: hidden;
  position: relative;

  &:hover > *{
    opacity: 1;
  }

  & ${LeftChavron}, & ${RightChavron}{
    pointer-events: none;

    ${media.large`
      pointer-events: all;
    `}
  }
`;

export const Body = styled.div`
  position: relative;
  --color: ${props => getTheme(props.theme, 'colors.v2.liberBlue', '#eaf7ff')};
`;
