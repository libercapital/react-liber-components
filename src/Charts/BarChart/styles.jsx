import styled from 'styled-components';
import { handleThemeFromObject } from 'Themes';

export const empty = null;

export const Slider = styled.input`
  width: ${props => (props.width ? `${props.width}px` : '100%')};
  appearance: none;
  height: 8.5px;
  background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.background', '#fff')};
  outline: none;
  border-radius: 4.25px;
  border: solid 2px ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};

  &::-webkit-slider-thumb {
    -webkit-appearance: none;
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &::-moz-range-thumb {
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &::-ms-thumb {
    border: 0px;
    width: 55px;
    height: 6px;
    border-radius: 3px;
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.border', '#dfe3eb')};;
  }

  &:focus {
    outline: none;
  }
  &::-moz-focus-outer {
    border: 0;
  }
  &::-moz-range-track {
    background-color: ${props => handleThemeFromObject(props.theme, 'barChart.slider.background', '#fff')};
  }
`;

Slider.defaultProps = {
  type: 'range',
  valueAsNumber: true,
};
