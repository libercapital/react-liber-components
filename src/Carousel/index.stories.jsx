import React from 'react';
import styled from 'styled-components';
import Carousel from './Carousel';

const BannerContainer = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 18px;
  box-sizing: border-box;
`;

export default {
  title: 'Components 1.0|Carousel',

  parameters: {
    jest: ['Carousel'],
  },
};

export const defaultStory = () => (
  <Carousel>
    <BannerContainer>Conteudo 1</BannerContainer>
    <BannerContainer>Conteudo 2</BannerContainer>
    <BannerContainer>Conteudo 3</BannerContainer>
  </Carousel>
);

defaultStory.story = {
  name: 'default',
};
