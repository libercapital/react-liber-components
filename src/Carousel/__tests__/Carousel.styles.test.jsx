import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import * as styles from '../Carousel.styles';

describe('Testing Carousel Styles', () => {
  configure({ adapter: new Adapter() });
  const { DotsContainer, Dot, ...uncontrolled } = styles;
  describe('Uncontroled', () => {
    Object.keys(uncontrolled).forEach((style) => {
      it(`renders ${style} without crashing`, () => {
        const App = styles[style];
        const wrapper = shallow(<App />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
  describe('Controlled', () => {
    describe('Dot', () => {
      it('should render without crashing, without current', () => {
        const wrapper = shallow(<DotsContainer />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('should render without crashing, with current=false', () => {
        const wrapper = shallow(<DotsContainer current={false} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('should render without crashing, with current=true', () => {
        const wrapper = shallow(<DotsContainer current />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
    describe('DotContainer', () => {
      it('should render without crashing, without inner', () => {
        const wrapper = shallow(<DotsContainer />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('should render without crashing, with inner=false', () => {
        const wrapper = shallow(<DotsContainer inner={false} />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('should render without crashing, with inner=true', () => {
        const wrapper = shallow(<DotsContainer inner />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
