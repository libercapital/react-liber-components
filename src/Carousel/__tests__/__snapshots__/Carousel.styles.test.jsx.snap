// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Carousel Styles Controlled Dot should render without crashing, with current=false 1`] = `
<StyledComponent
  current={false}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__DotsContainer-nmajyf-1",
        "isStatic": false,
        "rules": Array [
          "min-height:30px;width:100%;display:flex;justify-content:center;align-items:center;",
          [Function],
          ";",
        ],
      },
      "displayName": "Carouselstyles__DotsContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__DotsContainer-nmajyf-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Carousel Styles Controlled Dot should render without crashing, with current=true 1`] = `
<StyledComponent
  current={true}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__DotsContainer-nmajyf-1",
        "isStatic": false,
        "rules": Array [
          "min-height:30px;width:100%;display:flex;justify-content:center;align-items:center;",
          [Function],
          ";",
        ],
      },
      "displayName": "Carouselstyles__DotsContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__DotsContainer-nmajyf-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Carousel Styles Controlled Dot should render without crashing, without current 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__DotsContainer-nmajyf-1",
        "isStatic": false,
        "rules": Array [
          "min-height:30px;width:100%;display:flex;justify-content:center;align-items:center;",
          [Function],
          ";",
        ],
      },
      "displayName": "Carouselstyles__DotsContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__DotsContainer-nmajyf-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Carousel Styles Controlled DotContainer should render without crashing, with inner=false 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__DotsContainer-nmajyf-1",
        "isStatic": false,
        "rules": Array [
          "min-height:30px;width:100%;display:flex;justify-content:center;align-items:center;",
          [Function],
          ";",
        ],
      },
      "displayName": "Carouselstyles__DotsContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__DotsContainer-nmajyf-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  inner={false}
/>
`;

exports[`Testing Carousel Styles Controlled DotContainer should render without crashing, with inner=true 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__DotsContainer-nmajyf-1",
        "isStatic": false,
        "rules": Array [
          "min-height:30px;width:100%;display:flex;justify-content:center;align-items:center;",
          [Function],
          ";",
        ],
      },
      "displayName": "Carouselstyles__DotsContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__DotsContainer-nmajyf-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  inner={true}
/>
`;

exports[`Testing Carousel Styles Controlled DotContainer should render without crashing, without inner 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__DotsContainer-nmajyf-1",
        "isStatic": false,
        "rules": Array [
          "min-height:30px;width:100%;display:flex;justify-content:center;align-items:center;",
          [Function],
          ";",
        ],
      },
      "displayName": "Carouselstyles__DotsContainer",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__DotsContainer-nmajyf-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Carousel Styles Uncontroled renders Body without crashing 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__Body-nmajyf-6",
        "isStatic": false,
        "rules": Array [
          "position:relative;--color:",
          [Function],
          ";",
        ],
      },
      "displayName": "Carouselstyles__Body",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__Body-nmajyf-6",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Carousel Styles Uncontroled renders Carousels without crashing 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__Carousels-nmajyf-0",
        "isStatic": false,
        "rules": Array [
          "display:flex;transform:translateX(",
          [Function],
          ");transition:transform .5s ease-in-out;width:100%;height:100%;& > *{min-width:100%;min-height:100%;}",
        ],
      },
      "displayName": "Carouselstyles__Carousels",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__Carousels-nmajyf-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Carousel Styles Uncontroled renders Hidden without crashing 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__Hidden-nmajyf-5",
        "isStatic": false,
        "rules": Array [
          "--color:",
          [Function],
          ";width:",
          [Function],
          ";height:",
          [Function],
          ";overflow:hidden;position:relative;&:hover > *{opacity:1;}& ",
          ".Carouselstyles__LeftChavron-nmajyf-3",
          ",& ",
          ".Carouselstyles__RightChavron-nmajyf-4",
          "{pointer-events:none;",
          "@media (min-width:",
          "992px",
          "){",
          "
      pointer-events: all;
    ",
          "}",
          "}",
        ],
      },
      "displayName": "Carouselstyles__Hidden",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__Hidden-nmajyf-5",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Carousel Styles Uncontroled renders LeftChavron without crashing 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__LeftChavron-nmajyf-3",
        "isStatic": false,
        "rules": Array [
          "transform:rotate(90deg);left:10px;",
          "width:50px;height:50px;position:absolute;top:calc(50% - 25px);fill:var(--color);cursor:pointer;opacity:0.2;transition:opacity .25s ease-in-out;",
        ],
      },
      "displayName": "Carouselstyles__LeftChavron",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__LeftChavron-nmajyf-3",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Carousel Styles Uncontroled renders RightChavron without crashing 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Carouselstyles__RightChavron-nmajyf-4",
        "isStatic": false,
        "rules": Array [
          "transform:rotate(-90deg);right:10px;",
          "width:50px;height:50px;position:absolute;top:calc(50% - 25px);fill:var(--color);cursor:pointer;opacity:0.2;transition:opacity .25s ease-in-out;",
        ],
      },
      "displayName": "Carouselstyles__RightChavron",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Carouselstyles__RightChavron-nmajyf-4",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
